packages/asset-swapper/src/utils/swap_quote_calculator.ts

```diff
@@ -266,6 +279,49 @@ export class SwapQuoteCalculator {
             protocolFeeInWeiAmount,
         };
     }
+
+    private _determineLiquiditySource(order: SignedOrder): ERC20BridgeSource {
+        const decodedAssetData = assetDataUtils.decodeAssetDataOrThrow(order.makerAssetData);
+        if (decodedAssetData.assetProxyId === AssetProxyId.ERC20Bridge) {
+            const bridgeAddress = (decodedAssetData as ERC20BridgeAssetData).bridgeAddress;
+            switch (bridgeAddress) {
+                case this._contractAddresses.kyberBridge: return ERC20BridgeSource.Kyber;
+                case this._contractAddresses.uniswapBridge: return ERC20BridgeSource.Uniswap;
+                case this._contractAddresses.eth2DaiBridge: return ERC20BridgeSource.Eth2Dai;
+                default: return ERC20BridgeSource.Native; // HACK: should we throw an error for unknown?
+            }
+        }
+        return ERC20BridgeSource.Native;
+    }
+
+    private _getSwapQuoteOrdersBreakdown(
+        orders: SignedOrder[],
+        totalAssetAmount: BigNumber,
+        operation: MarketOperation,
+    ): SwapQuoteOrdersBreakdown {
+        // HACK: to shut up linter
+        const breakdown: SwapQuoteOrdersBreakdown = {};
+
+        const aggregatedFillAmountBySource: SwapQuoteOrdersBreakdown = orders.reduce((acc: SwapQuoteOrdersBreakdown, order: SignedOrder): SwapQuoteOrdersBreakdown => {
+            const assetAmount = operation === MarketOperation.Buy ? order.makerAssetAmount : order.takerAssetAmount;
+            const source = this._determineLiquiditySource(order);
+            return {
+                ...acc,
+                ...{
+                    [source]: !!acc[source] ? acc[source].plus(assetAmount) : assetAmount,
+                },
+            };
+        }, breakdown);
+
+        return Object.keys(aggregatedFillAmountBySource).reduce((acc: SwapQuoteOrdersBreakdown, source: string): SwapQuoteOrdersBreakdown => {
+            return {
+                ...{
+                    [source]: aggregatedFillAmountBySource[source].dividedBy(totalAssetAmount),
+                },
+                ...acc,
+            };
+        }, breakdown);
+    }
 }
```

Comment:
There's actually a more direct way to get this info.  `getMarketSellOrdersAsync()` now returns a `OptimizedMarketOrder[]` type, which each have a `fill` field that specifies the source and asset amount.

Reply:
<face palm>